src/comportamientos/PatearPelota.ts

```diff
@@ -0,0 +1,36 @@
+/// <reference path = "../../node_modules/pilasweb/dist/pilasweb.d.ts"/>
+/// <reference path = "Interactuar.ts" />
+
+class PatearPelota extends Interactuar {
+
+    protected alInteractuar(): void {
+        super.alInteractuar()
+        this.patearPelota()
+    }
+
+    /**
+     * Patea una pelota que este en la misma posición
+     * que el interactor del comportamiento.
+     */
+    private patearPelota(): void {
+        this.interactuado().hacer(SerPateado, {
+            tiempoEnElAire: 25,
+            aceleracion: 0.0025,
+            elevacionMaxima: 25,
+            gradosDeAumentoStep: -2
+        })
+    }
+
+    etiqueta(): string {
+        return "PelotaAnimada"
+    }
+
+    nombreAnimacion(): string {
+        return "patear"
+    }
+
+    hayConQuienInteractuar(): boolean {
+        return super.hayConQuienInteractuar() && !this.interactuado()['pateado']
+    }
```

Comment:
Yo no redefiniría este método. Por lo que menciono en el PR de Pilas Bloques sobre que esto tiene un problema. Explico un poco: como en Interactuar el código es este:

```Typescript
#Interactuar>>configurarVerificaciones(): void {
        const mensajeError: string = this.mensajeDeError() || "¡Acá no hay " + this.hacerLegible(this.etiqueta()) + "!"
        this.verificacionesPre.push(new Verificacion(() => this.hayConQuienInteractuar(), mensajeError))
    }
```

Lo que hace `hayConQuienInteractuar` es darme la pregunta para luego tirar el error "no hay pelota".

Lo que vos querés hacer es agregar otra verificación:


```suggestion
    configurarVerificaciones(): void {
        super.configurarVerificaciones()
        this.verificacionesPre.push(new Verificacion(() => !!this.interactuado().pateado, "No puedo patear dos veces la misma pelota"))
    }
```

Reply:
me gusta, me gusta